File: VB/dxSample_CustomSimpleButton/Custom Button/CustomSkinEditorButtonPainter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors.Drawing;
using DevExpress.Skins;
using DevExpress.Utils.Drawing;

namespace dxSample_CustomSimpleButton.Custom_Button {
    public class CustomSkinEditorButtonPainter : SkinEditorButtonPainter {
        public CustomSkinEditorButtonPainter(DevExpress.Skins.ISkinProvider provider)
            : base(provider) {
        }
        public CustomSkinEditorButtonPainter(DevExpress.Skins.ISkinProvider provider, object OwnerButton)
            : base(provider) {
            Button = OwnerButton as CustomButton;
        }
        private CustomButton Button;
        private CustomStyle GetCustomStyle() {
            CustomStyle style;
            if (Button != null) {
                style = Button.CustomStyle;
            }            else {
                style = CustomStyle.Style1;
            }
            return style;
        }
        protected override SkinElement GetSkinElement(EditorButtonObjectInfoArgs e, DevExpress.XtraEditors.Controls.ButtonPredefines kind) {
            return base.GetSkinElement(e, kind);
        }
        protected override int CalcImageIndexCore(DevExpress.Utils.Drawing.ObjectState state) {
            var customStyle = GetCustomStyle();
            var startImageIndex = 0;
            switch (customStyle) {
                case CustomStyle.Style1:
                    startImageIndex = 0;
                    break;
                case CustomStyle.Style2:
                    startImageIndex = 6;
                    break;
                case CustomStyle.Style3:
                    startImageIndex = 12;
                    break;
            }
            var tempState = state & (~ObjectState.Selected);
            if (tempState == ObjectState.Disabled) {
                return startImageIndex + 3;
            }
            if ((tempState & ObjectState.Pressed) != 0) {
                return startImageIndex + 2;
            }
            if ((tempState & ObjectState.Hot) != 0) {
                return startImageIndex + 1;
            }
            if ((state & ObjectState.Selected) != 0) {
                if (info.Element == null || info.Element.Image == null || info.Element.Image.ImageCount < 5) {
                    return startImageIndex + 0;
                }
                return startImageIndex + 4;
            }
            return startImageIndex + 0;
        }
    }
}

File: VB/dxSample_CustomSimpleButton/Custom Button/CustomSkinEditorButtonPainter.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.Skins
Imports DevExpress.Utils.Drawing

Namespace dxSample_CustomSimpleButton.Custom_Button
    Public Class CustomSkinEditorButtonPainter
        Inherits SkinEditorButtonPainter

        Public Sub New(ByVal provider As DevExpress.Skins.ISkinProvider)
            MyBase.New(provider)
        End Sub
        Public Sub New(ByVal provider As DevExpress.Skins.ISkinProvider, ByVal OwnerButton As Object)
            MyBase.New(provider)
            Button = TryCast(OwnerButton, CustomButton)
        End Sub
        Private Button As CustomButton
        Private Function GetCustomStyle() As CustomStyle
            Dim style As CustomStyle
            If Button IsNot Nothing Then
                style = Button.CustomStyle
            Else
                style = CustomStyle.Style1
            End If
            Return style
        End Function
        Protected Overrides Function GetSkinElement(ByVal e As EditorButtonObjectInfoArgs, ByVal kind As DevExpress.XtraEditors.Controls.ButtonPredefines) As SkinElement
            Return MyBase.GetSkinElement(e, kind)
        End Function
        Protected Overrides Function CalcImageIndexCore(ByVal state As DevExpress.Utils.Drawing.ObjectState) As Integer
            Dim customStyle = GetCustomStyle()
            Dim startImageIndex = 0
            Select Case customStyle
                Case dxSample_CustomSimpleButton.Custom_Button.CustomStyle.Style1
                    startImageIndex = 0
                Case dxSample_CustomSimpleButton.Custom_Button.CustomStyle.Style2
                    startImageIndex = 6
                Case dxSample_CustomSimpleButton.Custom_Button.CustomStyle.Style3
                    startImageIndex = 12
            End Select
            Dim tempState = state And ((Not ObjectState.Selected))
            If tempState = ObjectState.Disabled Then
                Return startImageIndex + 3
            End If
            If (tempState And ObjectState.Pressed) <> 0 Then
                Return startImageIndex + 2
            End If
            If (tempState And ObjectState.Hot) <> 0 Then
                Return startImageIndex + 1
            End If
            If (state And ObjectState.Selected) <> 0 Then
                If info.Element Is Nothing OrElse info.Element.Image Is Nothing OrElse info.Element.Image.ImageCount < 5 Then
                    Return startImageIndex + 0
                End If
                Return startImageIndex + 4
            End If
            Return startImageIndex + 0
        End Function
    End Class
End Namespace
